Gordon360/Controllers/ScheduleController.cs

```diff
@@ -68,37 +68,41 @@ public ActionResult<ScheduleViewModel> Get()
         /// <returns>A IEnumerable of schedule objects</returns>
         [HttpGet]
         [Route("{username}")]
-        public async Task<ActionResult<JArray>> GetAsync(string username)
+        public async Task<ActionResult<JArray>> GetAsync(string username, [FromQuery] string? sessionID)
         {
             //probably needs privacy stuff like ProfilesController and service
             var viewerGroups = AuthUtils.GetGroups(User);
 
+
+            var authenticatedUserUsername = AuthUtils.GetUsername(User);
+
             var groups = AuthUtils.GetGroups(username);
             var id = _accountService.GetAccountByUsername(username).GordonID;
             var scheduleControl = _context.Schedule_Control.Find(id);
 
             IEnumerable<ScheduleViewModel>? scheduleResult = null;
+            if (authenticatedUserUsername == username)
+            {
+                scheduleResult = await _scheduleService.GetScheduleStudentAsync(username, sessionID);
+            }
+
             if (groups.Contains(AuthGroup.Student))
```

Comment:
This needs to be a connected conditional chain (i.e. make the second if an else if). Otherwise, `scheduleResult` will be overwritten.

Reply:
We just pushed the changes made for this and added this condition as the first if statement in the `if (groups.Contains(AuthGroup.Student))` statement. We also removed the variables for `schedulePrivacy` because students are not allowed to see other students' schedules anyways (and it is deprecated in the frontend as well).